### ISSUE TITLE
Question Regarding Equation Implementation in the Paper

### ISSUE THREAD

**Issue by rlawjdghek** [wfanyue/DPG-T2I-Personalization#7]
Hello, thank you for the great paper — I really enjoyed reading it.
While comparing the algorithm and the code, I noticed that in line 9 of Algorithm 1, the diffusion model performs gradient ascent with respect to q_phi_reward.
However, in the code, it seems to be implemented as gradient descent.
Could you please clarify the reason for this discrepancy?

**Comment by wfanyue** [comment]
Hi,

Thank you very much for your interest in our work and for your insightful question that helps clarify potential points of confusion.

Our RL algorithm (Deterministic Policy Gradient) trains the diffusion model by maximizing the reward; we follow the standard DDPG pipeline (https://spinningup.openai.com/en/latest/algorithms/ddpg.html) in Algorithm 1 in the paper. In implementation, maximizing the reward is equivalent to minimizing its negation. Concretely, we treat the reward as **r(·)= -Loss(·)**, 
then we can optimize the model using the standard PyTorch gradient descent optimizer. 

If you have any further questions, feel free to reach.

Best
Fanyue

**Comment by rlawjdghek** [comment]
Thank you for your response. But where exactly in the code is the part where the negative sign is applied to the reward? Is it from around line 1347 in train_dreambooth_dpg.py?

**Comment by wfanyue** [comment]
Sorry for not clarifying clearly, so I modified the answer above. The reward is `torch.square(x_0_gt - x_0_latents)` at line 1346, so minimizing it is equivalent to maximizing its negation, i.e., **-Loss(·)**. 
For the code implementation of our DPG method, you can refer to lines 1346 to 1380 in `train_dreambooth_dpg.py`. The other lengthy part of this script is directly reused from the diffusers code.
